docs/docs/basic/config.md
---
title: Configuration
date: 2021-02-03
---

The theme can be configured in `.vuepress/config.js`. Refer to [here](https://github.com/Renovamen/vuepress-theme-gungnir/blob/main/packages/theme/src/shared/options.ts) for all possible options.

::: tip
The configs of this theme are compatible with all the configs of [VuePress's default theme](https://v2.vuepress.vuejs.org/reference/default-theme/config.html) except:

- logo
- logoDark
- darkMode (dark mode switching is always enabled in this theme)
- repo
- repoLabel

Please refer to [reference of VuePress's default theme config](https://v2.vuepress.vuejs.org/reference/default-theme/config.html) for how to set them.
:::

::: tip
Please make sure all the icons you specified in the configuration file are already imported, see [here](/docs/advanced/icons/) to know how to import icons.
:::


## Navbar Title

Different from the default theme, Gungnir theme displays `themeConfig.navbarTitle` instead of the [`title` option in site config](https://v2.vuepress.vuejs.org/reference/config.html#title) at the left of the navbar:


```js{14}
// .vuepress/config.js

module.exports = {
  /*
    other site configurations
    ...
  */

  theme: gungnirTheme({
    /*
      other theme configurations
      ...
    */
    navbarTitle: "Gungnir"  // optional, default: "$ cd /home/"
  })
}
```

It can also be used in the [`locales`](https://v2.vuepress.vuejs.org/reference/default-theme/config.html#locales) option, for example:

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    locales: {
      "/": {
        navbarTitle: "Gungnir"
      },
      "/zh/": {
        navbarTitle: "冈格尼尔"
      }
    }
  })
}
```


## Author Information

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    personalInfo: {
      // required: nickname, will be showed on home page, mobile sidebar and author info of articles
      name: "Cool Dragon",

      // required: avatar, will be showed on home page and mobile sidebar
      avatar: "/img/avatar.jpeg",

      // required: a description, will be showed on home page
      description: 'A cool dragon lost in human world.',
      
      // optional: id of social platform, will be showed on home page and mobile sidebar
      sns: {
        github: 'Renovamen',  // Github
        linkedin: 'xiaohan-zou-55bba0160',  // Linkedin
        facebook: 'renovamen.zou',  // Facebook
        twitter: 'renovamen_zxh',  // Twitter
        zhihu: 'chao-neng-gui-su',  // Zhihu
        weibo: 'your-weibo-id',  // Sina Weibo
        email: 'renovamenzxh@gmail.com',  // Email
        rss: "/rss.xml",  // link to RSS file
        // if you want to add other social platforms
        bilibili: {  // any name you want
          icon: "ri-bilibili-line",  // platform icon
          link: "https://www.bilibili.com/"  // profile URL
        }
        // ...
      }
    }
  })
}
```

## Home Page

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    homeHeaderImages: [
      // image 1
      {
        "path": "/img/home-bg/1.jpg",
        "mask": "rgba(40, 57, 101, .4)"
      },
      // image 2
      {
        "path": "/img/home-bg/2.jpg",
        "mask": "rgb(251, 170, 152, .2)"
      }
      ...
    ],
    blogNumPerPage: 20  // optional: maximum number of blogs per page, default: 10
  })
}
```

The `mask` is a translucent overlay placed on the cover image. It will be helpful when your cover image is so complex that your title can not stand out.

Check the [home page](/) of this site to see the effect. Click the left and right button to switch among multiple cover images.


## Other Pages

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    pages: {
      // tags page
      tags: {
        // optional: subtitle of tags page
        subtitle: 'Hey! Here is tags page!',

        // optional: paths to and masks of the tags page cover images
        bgImage: {
          path: '/img/pages/tags.jpg',
          mask: 'rgba(211, 136, 37, .5)'
        }
      },

      // links page
      links: {
        // optional: subtitle of links page
        subtitle: 'Hey! Here is links page!',

        // optional: paths to and masks of the links page cover images
        bgImage: {
          path: '/img/pages/links.jpg',
          mask: 'rgba(64, 118, 190, 0.5)'
        }
      }
    }
  })
}
```

Check the [tags page](/tags/) and [links page](/links/) to see the effect.

To edit the title of the above mentioned pages, please refer to Multi-language Support (TODO) section.


## Custom Footer

Support HTML. For example, the footer of this site is:

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    footer: `
      &copy; <a href="https://github.com/Renovamen" target="_blank">Renovamen</a> 2018-2022
      <br>
      Powered by <a href="https://v2.vuepress.vuejs.org" target="_blank">VuePress</a> &
      <a href="https://github.com/Renovamen/vuepress-theme-gungnir" target="_blank">Gungnir</a>
    `
  })
}
```


## Catalog

Enable or disable **post** catalog for all posts. Can be rewritten in each post's front matter.

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    catalog: false  // optional, default: true
  })
}
```


docs/zh/docs/basic/config.md
---
title: 基本配置
date: 2021-02-03
---

在 `.vuepress/config.js` 中配置主题。可以直接在[源码里](https://github.com/Renovamen/vuepress-theme-gungnir/blob/main/packages/theme/src/shared/options.ts)查看所有可能的配置项。

::: tip
本主题兼容 [VuePress 默认主题](https://v2.vuepress.vuejs.org/zh/reference/default-theme/config.html)除了以下配置项以外的所有配置项：

- logo
- logoDark
- darkMode（本主题强制开启模式切换功能）
- repo
- repoLabel

关于这些（兼容的）配置项的配置方式，请直接查阅 [VuePress 默认主题配置的参考文档](https://v2.vuepress.vuejs.org/zh/reference/default-theme/config.html)。
:::

::: tip
对于所有涉及到图标名称的配置项，请确保指定的图标是被引入过的，在[这里](/zh/docs/advanced/icons/)查看怎么引入图标。
:::


## 导航栏标题

与默认主题不同的是，本主题的导航栏左上角标题处显示的不是 [站点配置中的 `title` 项](https://v2.vuepress.vuejs.org/zh/reference/config.html#title)，而是：

```js{14}
// .vuepress/config.js

module.exports = {
  /*
    其他站点配置
    ...
  */

  theme: gungnirTheme({
    /*
      其他主题配置
      ...
    */
    navbarTitle: "Gungnir"  // 可选，默认："$ cd /home/"
  })
}
```

也可以在 [`locales`](https://v2.vuepress.vuejs.org/zh/reference/default-theme/config.html#locales) 中使用，如：

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    locales: {
      "/": {
        navbarTitle: "Gungnir"
      },
      "/zh/": {
        navbarTitle: "冈格尼尔"
      }
    }
  })
}
```


## 作者信息

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    personalInfo: {
      // 必须：名称，将在首页、移动端侧边栏和文章作者信息处显示
      name: "Cool Dragon",

      // 必须：头像，将在首页和移动端侧边栏显示
      avatar: "/img/avatar.jpeg",

      // 必须：个人简介，将在首页显示
      description: "A cool dragon lost in human world.",
      
      // 可选：社交平台账号，将在首页和移动端侧边栏显示
      sns: {
        github: "Renovamen",  // Github
        linkedin: "xiaohan-zou-55bba0160",  // 领英
        facebook: "renovamen.zou",  // Facebook
        twitter: "renovamen_zxh",  // 推特
        zhihu: "chao-neng-gui-su",  // 知乎
        weibo: "your-weibo-id",  // 新浪微博
        email: "renovamenzxh@gmail.com",  // 邮箱
        rss: "/rss.xml",  // RSS 文件的链接
        // 添加其他的社交平台
        bilibili: {  // 随便什么名字
          icon: "ri-bilibili-line",  // 社交平台的图标
          link: "https://www.bilibili.com/"  // 主页链接
        }
        // ...
      }
    }
  })
}
```

## 首页

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    homeHeaderImages: [
      // 图 1
      {
        "path": "/img/home-bg/1.jpg",
        "mask": "rgba(40, 57, 101, .4)"
      },
      // 图 2
      {
        "path": "/img/home-bg/2.jpg",
        "mask": "rgb(251, 170, 152, .2)"
      }
      ...
    ],
    blogNumPerPage: 20  // 可选：每页的博客数量，默认：10
  })
}
```

其中，蒙版是指在封面图上盖一层半透明的颜色，如果图片比较复杂导致文字显示得不够清楚，可以考虑加个蒙版。

页面效果可以参考本站的[首页](/)，点击封面图左右两边的按钮可以在多张图片之间进行切换。


## 页面

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    pages: {
      // 标签页配置
      tags: {
        // 可选：标签页副标题
        subtitle: '吼哇朋友们，这是标签页',

        // 可选：标签页封面图路径和蒙版
        bgImage: {
          path: '/img/pages/tags.jpg',
          mask: 'rgba(211, 136, 37, .5)'
        }
      },

      // 链接页配置
      links: {
        // 可选：链接页副标题
        subtitle: '吼哇朋友们，这是链接页',

        // 可选：链接页封面图路径和蒙版
        bgImage: {
          path: '/img/pages/links.jpg',
          mask: 'rgba(64, 118, 190, 0.5)'
        }
      }
    }
  })
}
```

页面效果可以参考本站的[标签](/tags)和[链接](/links)页。

如果想要修改页面主标题，请参考多语言支持部分（在写了在写了咕咕咕）。


## 页脚自定义

页脚支持 HTML 格式，如本站的页脚配置为：

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    footer: `
      &copy; <a href="https://github.com/Renovamen" target="_blank">Renovamen</a> 2018-2022
      <br>
      Powered by <a href="https://v2.vuepress.vuejs.org" target="_blank">VuePress</a> &
      <a href="https://github.com/Renovamen/vuepress-theme-gungnir" target="_blank">Gungnir</a>
    `
  })
}
```


## 目录

对所有**博客文章页**启用或禁用目录。可以在博客文章页的 front matter 里覆写这个配置。

```js
// .vuepress/config.js

module.exports = {
  theme: gungnirTheme({
    catalog: false  // 可选，默认：true
  })
}
```


CHANGELOG.md
# [2.0.0-alpha.23](https://github.com/Renovamen/vuepress-theme-gungnir/compare/v2.0.0-alpha.22...v2.0.0-alpha.23) (2022-05-20)


### IMPORTANT

To be compatible with VuePress >= 2.0.0-beta.40, this release contains important breaking changes, see [here](https://github.com/vuepress/vuepress-next/blob/main/CHANGELOG.md#200-beta40-2022-04-25) for details.

You could migrate your config file as follow, and check our latest docs for more details:

```diff
- module.exports = {
-   theme: "gungnir",
-   themeConfig: {
-     // your theme configs
-   }
- }

+ const { gungnirTheme } = require("vuepress-theme-gungnir");
+
+ module.exports = {
+   theme: gungnirTheme({
+     // you theme configs
+   })
+ }
```

```diff
- module.exports = {
-   plugins: [
-     [
-       "vuepress-plugin-giscus",
-       {
-         repo: "[repo]",
-         repoId: "[repo id]",
-         category: "[category name]",
-         categoryId: "[category id]"
-       }
-     ]
-   ]
- }

+ const { giscusPlugin } = require("vuepress-plugin-giscus");
+
+ module.exports = {
+   plugins: [
+     giscusPlugin({
+         repo: "[repo]",
+         repoId: "[repo id]",
+         category: "[category name]",
+         categoryId: "[category id]"
+     })
+   ]
+ }
```


### Bug Fixes

* **plugin-chart:** replace eval() with safer Function() ([ffd7913](https://github.com/Renovamen/vuepress-theme-gungnir/commit/ffd7913335971bc033e7234552065a39d72546c2))


### Features

* improve SEO (render blog list on server side) [#62](https://github.com/Renovamen/vuepress-theme-gungnir/issues/62) [#28](https://github.com/Renovamen/vuepress-theme-gungnir/issues/28) ([419b2c8](https://github.com/Renovamen/vuepress-theme-gungnir/commit/419b2c82169d2337283055ed27db2bda13710834))





# [2.0.0-alpha.22](https://github.com/Renovamen/vuepress-theme-gungnir/compare/v2.0.0-alpha.21...v2.0.0-alpha.22) (2022-05-01)


### Bug Fixes

* catalog height [#60](https://github.com/Renovamen/vuepress-theme-gungnir/issues/60) ([e3483b2](https://github.com/Renovamen/vuepress-theme-gungnir/commit/e3483b2ed3d97562cdb91e4fb71e62459d3c7aa0))





# [2.0.0-alpha.21](https://github.com/Renovamen/vuepress-theme-gungnir/compare/v2.0.0-alpha.20...v2.0.0-alpha.21) (2022-04-18)


### Bug Fixes

* **plugin-giscus:** fix giscus lang options setting, close [#57](https://github.com/Renovamen/vuepress-theme-gungnir/issues/57) ([#58](https://github.com/Renovamen/vuepress-theme-gungnir/issues/58)) ([bb200e2](https://github.com/Renovamen/vuepress-theme-gungnir/commit/bb200e230ef2179d96ea4fec7f0d41d850994e70))
* pwa default option ([a3d680a](https://github.com/Renovamen/vuepress-theme-gungnir/commit/a3d680ad28e280e5971e5b8785a9e1e4021a8307))





# [2.0.0-alpha.20](https://github.com/Renovamen/vuepress-theme-gungnir/compare/v2.0.0-alpha.19...v2.0.0-alpha.20) (2022-04-13)


### Bug Fixes

* compatible with vuepress v2.0.0-beta.39 [#55](https://github.com/Renovamen/vuepress-theme-gungnir/issues/55) ([e68d31b](https://github.com/Renovamen/vuepress-theme-gungnir/commit/e68d31b404cea050a0ca8847d99dfa4c7244498f))
* link-card title and icon name clash. error replace ([#50](https://github.com/Renovamen/vuepress-theme-gungnir/issues/50)) ([9db4cda](https://github.com/Renovamen/vuepress-theme-gungnir/commit/9db4cda145fbf3fc6998e0d0bb83d4c78b908c2f)), closes [#49](https://github.com/Renovamen/vuepress-theme-gungnir/issues/49) [#49](https://github.com/Renovamen/vuepress-theme-gungnir/issues/49)


### Features

* **plugin-rss:** add author, get title and desc from theme config ([0388ed2](https://github.com/Renovamen/vuepress-theme-gungnir/commit/0388ed2c4fde666b8ba70ba9e902ab7e410913d4))
* update link container styles ([dd9eb74](https://github.com/Renovamen/vuepress-theme-gungnir/commit/dd9eb74f8f0342838a616683b6a9b0836528c4b2))





# [2.0.0-alpha.19](https://github.com/Renovamen/vuepress-theme-gungnir/compare/v2.0.0-alpha.18...v2.0.0-alpha.19) (2022-03-29)


### Bug Fixes

* remove outlines of headings and sidebar button ([f8d7d49](https://github.com/Renovamen/vuepress-theme-gungnir/commit/f8d7d49b8c36c8f368e5d45a8fa9c16afcef8875))


### Features

* add link card container (close [#36](https://github.com/Renovamen/vuepress-theme-gungnir/issues/36)) ([#38](https://github.com/Renovamen/vuepress-theme-gungnir/pull/38)) ([415e5a2](https://github.com/Renovamen/vuepress-theme-gungnir/commit/415e5a2ccfac337f08cdb44d9cd63f5f41acc4b3))
* allows to hide site domain in link card container ([934ecae](https://github.com/Renovamen/vuepress-theme-gungnir/commit/934ecaed0e4e2efdab64de4edb2282f82e80eff5))
* **plugin-giscus:** added support for lazy load of the comments ([#45](https://github.com/Renovamen/vuepress-theme-gungnir/pull/45)) ([fecc326](https://github.com/Renovamen/vuepress-theme-gungnir/commit/fecc326ed91915d97153fbff6227ad4a5514b6c0))
* **plugin-chart:** support custom code block token ([0a48ac7](https://github.com/Renovamen/vuepress-theme-gungnir/commit/0a48ac7a6210e21b9d3ff6a4cbbc15e1e78dc43e))
* **plugin-mermaid:** support custom code block token [#46](https://github.com/Renovamen/vuepress-theme-gungnir/issues/46) ([54b6a00](https://github.com/Renovamen/vuepress-theme-gungnir/commit/54b6a003f3550d082f513ddcd8bba82ca473f24d))
* support crediting the source of the header image in blog ([9608ccf](https://github.com/Renovamen/vuepress-theme-gungnir/commit/9608ccf568c851593c67e4611ed17c2d2ca078f8))
* support custom navbar brand title [#43](https://github.com/Renovamen/vuepress-theme-gungnir/issues/43) ([4ef34d1](https://github.com/Renovamen/vuepress-theme-gungnir/commit/4ef34d1d900491d84b9dca82ddea8378e02e5d29))


### Performance Improvements

* don't fetch hitokoto util hovering avatar ([487c439](https://github.com/Renovamen/vuepress-theme-gungnir/commit/487c439db5c0d5d61ea97beb7dc36f92dcc13009))





# [2.0.0-alpha.18](https://github.com/Renovamen/vuepress-theme-gungnir/compare/v2.0.0-alpha.17...v2.0.0-alpha.18) (2022-03-19)


### Bug Fixes


example/.vuepress/config.js
module.exports = {
  title: "vuepress-theme-gungnir",
  description: "Gungnir is a blog theme for VuePress.",
  head: [
    ["link", { rel: "icon", href: "/img/logo.svg" }],
    ["meta", { name: "viewport", content: "width=device-width,initial-scale=1,user-scalable=no" }]
  ],
  theme: "gungnir",
  locales: {
    "/": {
      lang: "en-US"
    },
    "/zh/": {
      lang: "zh-CN"
    }
  },
  themeConfig: {
    repo: "Renovamen/vuepress-theme-gungnir",
    docsDir: "example",
    editLinks: true,
    lastUpdated: true,
    hitokoto: true,  // enable hitokoto (一言) or not?
    searchIcon: "ri-search-2-line",
    languageIcon: "hi-translate",
    rss: {
      site_url: "https://vuepress-theme-gungnir.vercel.app",
      copyright: "Renovamen 2018-2021",
      count: 20
    },
    comment: {
      owner: "This-is-an-Apple",
      repo: "gitalk-comments",
      clientId: "0a24dcc429074a13836a",
      clientSecret: "8d805542796c114a131012d2850a1e4d43772d98"
    },
    analytics: {
      ga: "UA-146858305-4",
      ba: "0958eaa31f4f4656f36bd33673332939"
    },
    katex: true,
    mermaid: true,
    chartjs: true,
    roughviz: true,
    markmap: true,
    mdPlus: {
      all: true
    },
    locales: {
      "/": {
        label: "English",
        selectText: "Languages",
        nav: require("./configs/nav/en"),
        sidebar: require("./configs/sidebar/en")
      },
      "/zh/": {
        label: "简体中文",
        selectText: "选择语言",
        nav: require("./configs/nav/zh"),
        sidebar: require("./configs/sidebar/zh")
      }
    },
    personalInfo: {
      name: "Gungnir",
      avatar: "/img/avatar.jpeg",
      description: "A blog theme for VuePress.",
      sns: {
        github: "Renovamen",
        linkedin: "xiaohan-zou-55bba0160",
        facebook: "renovamen.zou",
        twitter: "renovamen_zxh",
        zhihu: "chao-neng-gui-su",
        email: "renovamenzxh@gmail.com"
      }
    },
    homeHeaderImages: {
      local: [
        {
          path: "/img/home-bg/1.jpg",
          mask: "rgba(40, 57, 101, .4)"
        },
        {
          path: "/img/home-bg/2.jpg",
          mask: "rgb(251, 170, 152, .2)"
        },
        {
          path: "/img/home-bg/3.jpg",
          mask: "rgba(68, 74, 83, .1)"
        },
        {
          path: "/img/home-bg/4.jpg",
          mask: "rgba(19, 75, 50, .2)"
        }
      ]
    },
    pages: {
      tags: {
        subtitle: "Black Sheep Wall",
        bgImage: {
          path: "/img/pages/tags.jpg",
          mask: "rgba(211, 136, 37, .5)"
        }
      },
      links: {
        subtitle:
          "When you are looking at the stars, please put the brightest star shining night sky as my soul.",
        bgImage: {
          path: "/img/pages/links.jpg",
          mask: "rgba(64, 118, 190, 0.5)"
        }
      }
    },
    footer: `
      &copy; <a href="https://github.com/Renovamen" target="_blank">Renovamen</a> 2018-2021
      <br>
      Powered by <a href="https://vuepress.vuejs.org" target="_blank">VuePress</a> &
      <a href="https://github.com/Renovamen/vuepress-theme-gungnir" target="_blank">Gungnir</a>
    `
  },
  markdown: {
    // lineNumbers: true,
    extractHeaders: ["h2", "h3", "h4", "h5"]
  }
};
